### ISSUE TITLE
Lightning timestamp incorrect in Homey

### ISSUE THREAD

**Issue by EricMuhren** [AdyRock/com.misol#40]
Hi,

Very nice app! In Homey the timestamp of lightnings is one hour early. The time and date in the hub are correct. Is there a possibility to set an offset? Or another solution?

Thanks in advance.

**Comment by AdyRock** [comment]
Hmm, I'm not sure why. The time of my last strike reads the same in Homey and WS View. I can't say for sure if that was the correct time as it was 4 days ago and I didn't look then.
The app should adjust the UTC time to the same time zone as Homey, so is homey set to the correct region?

**Comment by EricMuhren** [comment]
Hi,

Today thunder again! I'll show you the screenshots. 


![_GW1100C](https://github.com/user-attachments/assets/4b2c498c-6dd3-429f-b19c-1df5823142bc)
![_Homey](https://github.com/user-attachments/assets/606e77de-a08e-4480-93a6-4c60676071ef)

**Comment by AdyRock** [comment]
I have a feeling that both the bridge and Homey are adjusting the time. 
I will add an option to switch off the adjustment in the device advanced settings.

**Comment by EricMuhren** [comment]
That would be nice!

**Comment by AdyRock** [comment]
Should be in the test version in the store.

**Comment by EricMuhren** [comment]
Thanks. I still have to wait for a thunderstorm to check.

**Comment by EricMuhren** [comment]
The datetime stamp in GW1100C is correct. 

With new option set no the difference is -2 hours. With option set to yes the difference is -1 hour.

**Comment by AdyRock** [comment]
Could you set the time format in the advanced settings to Time Stamp and the Adjust Time to No, then let me know what time is shown in the Homey device and what time is shown in WS View.
Then set the Adjust Time to Yes and let me know the Homey device time.

For me the time reported by the API (Time Stamp and the Adjust Time to No) is GMT 2024-08-21T00:05:29.000Z, while WS View is showing BST, 2024-08-21T01:05:29.000Z. If I set the Adjust time to Yes then both show BST.

What is your local time offset from GMT?
Do your clocks adjust for summer time?

**Comment by EricMuhren** [comment]
Hi, 

Time Stamp No
![TimeStampNo](https://github.com/user-attachments/assets/64c6abd5-6e8f-4b59-bb2c-4034310391df)

Time Stamp Yes
![TimeStampYes](https://github.com/user-attachments/assets/ac5c5220-b5cf-47c9-a319-354c48abed20)

Time Stamp WS
![TimeStampWS](https://github.com/user-attachments/assets/beb86afd-17fe-44a6-9311-b3f7dc3bb6c5)

Amsterdam = GMT + 1
Clocks are adjusted for summer time. (DST)

**Comment by AdyRock** [comment]
So just to make sure I understand, your current time is GMT + 2?

In which case it seems the API is returning GMT as expect but the app is not adjusting for your timezone.
The new option is switching off the summer time adjustments and that seems to be working as expected.

I am going to add a new log message to show what is being returned by Homey for the timezone value the app uses and I will let you know when that is ready.

**Comment by EricMuhren** [comment]
Yes, you're right. Timezone at this moment is GMT+2. Plus 1 in winter, plus 2 in summer.

**Comment by AdyRock** [comment]
I have published a test version that logs the timezone and language reported by Homey. That's the data I use in the time conversion, so would be interested to see what you have set.

Which insights?
